src/Services/SubmitService/SubmitService.cs

```diff
@@ -79,21 +87,54 @@ public async Task<string> ProcessSubmission(MappingEntity mappingEntity, string
                 reference = answers.CaseReference;
             }
 
-            return _submissionServiceConfiguration.FakeSubmission 
-                ? ProcessFakeSubmission(mappingEntity, form, sessionGuid, reference) 
+            var submissionReference = _submissionServiceConfiguration.FakeSubmission
+                ? ProcessFakeSubmission(mappingEntity, form, sessionGuid, reference)
                 : await ProcessGenuineSubmission(mappingEntity, form, sessionGuid, baseForm, reference);
+
+            if (baseForm.Pages is not null && mappingEntity.FormAnswers.Pages is not null)
+            {
+                var journeyPages = baseForm.GetReducedPages(mappingEntity.FormAnswers);
+
+                var isAutoConfirm = journeyPages.Any(_ => _.Elements.Any(_ => _.Type.Equals(EElementType.Booking) && _.Properties.AutoConfirm.Equals(true)));
+                if (isAutoConfirm)
+                    await ConfirmBooking(mappingEntity, form, sessionGuid, baseForm, reference);
```

Comment:
half of the paramters are not used and can be removed from the method sig, as we are not errogin we do not need to await this call.